python/cudf/cudf/core/column/column.py

```diff
@@ -616,22 +616,27 @@ def find_first_value(self, value):
         """
         Returns offset of first value that matches
         """
-        # FIXME: Inefficient find in CPU code
-        arr = self.to_array()
-        indices = np.argwhere(arr == value)
+        from cudf.core.index import RangeIndex
+
+        # FIXME: Inefficient, may be need a libcudf api
+        index = RangeIndex(0, stop=len(self))
+        indices = index.take(self == value)
         if not len(indices):
             raise ValueError("value not found")
-        return indices[-1, 0]
+        return indices[0]
```

Comment:
It really stinks to have to materialize a 0 --> N column here just to get the first `True` value from the elementwise equals op. Could we use an `argmax` instead here? Something like:
```
return (self == value).argmax()
```
That should return the idx of the first True value which can be used to grab the corresponding index which could be used to grab a scalar from the Index column if needed.

Reply:
I imagine there may be a better libcudf function already available to accomplish this though.

Reply:
Don't think we have any such API suport in lubcudf.